nodejs/eks/cluster.ts

```diff
@@ -229,6 +229,22 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
         throw new Error("subnetIds, and the use of publicSubnetIds and/or privateSubnetIds are mutually exclusive. Choose a single approach.");
     }
 
+    if (args.nodeGroupOptions && (
+        args.nodeSubnetIds ||
+        args.nodeAssociatePublicIpAddress ||
+        args.instanceType ||
+        args.instanceProfileName ||
+        args.nodePublicKey ||
+        args.nodeRootVolumeSize ||
+        args.nodeUserData ||
+        args.minSize ||
+        args.maxSize ||
+        args.desiredCapacity ||
+        args.nodeAmiId ||
+        args.version)) {
```

Comment:
Indirectly it is, since both the control plane and workers can differ in k8s versions used. In the node group we use `args.version` to form the `amiId` for the `nodeLaunchConfiguration`.

Though, after giving it some more thought I'm of the thinking that both of these can and should be configurable, so I'll remove this line from the mutual exclusivity check. Thanks for spotting it!